### ISSUE TITLE
Migrate to new getDisplayMedia API

### ISSUE THREAD

**Issue by saghul** [jitsi/jitsi-meet-electron-sdk#277]
Once released: https://github.com/electron/electron/pull/30702

**Comment by csett86** [comment]
Should be part of Electron 22 (as electron 21 was branched before the merge and the PR is not intended to be backported), so somewhere end of nov, beginning of dev 2022

**Comment by csett86** [comment]
Going forward this may help to reduce the double permission dialogs on wayland once a solution discussed here is present: https://github.com/electron/electron/issues/30652#issuecomment-1732848860

**Comment by damencho** [comment]
Hey @csett86 and @saghul  what do you think is missing here for using getDisplayMedia? There is a property now in lib-jitsi-meet to use it. But when I try I get the following error:
```
getDisplayMedia error {"video":{"height":99999,"width":99999},"audio":true,"preferCurrentTab":false,"systemAudio":"include","surfaceSwitching":"include","selfBrowserSurface":"exclude","cursor":"always"} {"errorName":"NotSupportedError","errorMsg":"Not supported"}
```
You can test it by using alpha.jitsi.net, just pass #config.testing.electronUseGetDisplayMedia=true to the room link.

**Comment by saghul** [comment]
We need to implement the handling in the SDK. It still requires us to show the picker.

**Comment by damencho** [comment]
But isn't that there is a native picker now ... it wasn't obvious to me what needs to be done when searching around.

**Comment by saghul** [comment]
Yeah it weird. No native picker. See: https://electronjs.org/docs/latest/api/desktop-capturer

**Comment by csett86** [comment]
Only for mac, and only for macOS 15+ there is a experimental possibility to use a system picker, but not for any other platform, unfortunately.

Esp. Linux/Wayland would benefit from the system picker (as its currently double picker there), but for older macOS and Windows and X11 we would need to maintain our own picker UI for quite a while

**Comment by saghul** [comment]
Indeed, we should detect and use the system picker when we can.
